File: sdk/prady-sdk/src/fs.ts
async function readJson<T>(response: Response): Promise<T> {
  if (!response.ok) {
    const detail = await response.text();
    throw new Error(detail || `HTTP ${response.status}`);
  }
  return (await response.json()) as T;
}

export class KryosFS {
  constructor(
    private readonly appName: string,
    private readonly apiBase: string = "http://localhost:8001"
  ) {}

  async read(relativePath: string): Promise<string> {
    const response = await fetch(
      `${this.apiBase}/sdk/fs/read?app=${encodeURIComponent(this.appName)}&path=${encodeURIComponent(relativePath)}`
    );
    const data = await readJson<{ content: string }>(response);
    return data.content;
  }

  async write(relativePath: string, content: string): Promise<void> {
    const response = await fetch(`${this.apiBase}/sdk/fs/write`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ app: this.appName, path: relativePath, content }),
    });
    await readJson<{ written: boolean }>(response);
  }

  async list(relativePath: string): Promise<string[]> {
    const response = await fetch(
      `${this.apiBase}/sdk/fs/list?app=${encodeURIComponent(this.appName)}&path=${encodeURIComponent(relativePath)}`
    );
    const data = await readJson<{ entries?: Array<{ name: string }> } | string[]>(response);
    if (Array.isArray(data)) {
      return data;
    }
    return (data.entries ?? []).map((entry) => entry.name);
  }

  async delete(relativePath: string): Promise<void> {
    const response = await fetch(
      `${this.apiBase}/sdk/fs/delete?app=${encodeURIComponent(this.appName)}&path=${encodeURIComponent(relativePath)}`,
      { method: "DELETE" }
    );
    await readJson<{ deleted: boolean }>(response);
  }
}

File: sdk/prady-sdk-python/kryos_sdk/fs.py
from __future__ import annotations

import httpx


class KryosFS:
    def __init__(self, app_name: str, api_base: str = "http://localhost:8001") -> None:
        self.app_name = app_name
        self.api_base = api_base

    async def read(self, relativePath: str) -> str:
        async with httpx.AsyncClient() as client:
            resp = await client.get(f"{self.api_base}/sdk/fs/read", params={"app": self.app_name, "path": relativePath})
        resp.raise_for_status()
        return resp.json()["content"]

    async def write(self, relativePath: str, content: str) -> None:
        async with httpx.AsyncClient() as client:
            resp = await client.post(f"{self.api_base}/sdk/fs/write", json={"app": self.app_name, "path": relativePath, "content": content})
        resp.raise_for_status()

    async def list(self, relativePath: str) -> list[str]:
        async with httpx.AsyncClient() as client:
            resp = await client.get(f"{self.api_base}/sdk/fs/list", params={"app": self.app_name, "path": relativePath})
        resp.raise_for_status()
        data = resp.json()
        if isinstance(data, list):
            return data
        return [entry["name"] for entry in data.get("entries", [])]

    async def delete(self, relativePath: str) -> None:
        async with httpx.AsyncClient() as client:
            resp = await client.delete(f"{self.api_base}/sdk/fs/delete", params={"app": self.app_name, "path": relativePath})
        resp.raise_for_status()
